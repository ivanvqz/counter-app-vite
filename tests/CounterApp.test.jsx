const { render, screen, fireEvent } = require("@testing-library/react")
const { CounterApp } = require("../src/CounterApp")


describe('Pruebas en el <CounterApp/>', () => { 
    const initialValue = 20

    test('debe de hacer match con el snapshot', () => { 
        const { container } = render( <CounterApp value={ initialValue }/> )
        expect( container ).toMatchSnapshot()
    })

    test('Debe mostrar el valor inicual de 100', () => { 
        render( <CounterApp value={ 100 }/> )
        //se espera que el elemento con el id counter tenga un valor de 100
        expect( screen.getByText(100)).toBeTruthy()
    }) 

    test('Debe de incrementar el botón +1', () => { 
        const { container } = render( <CounterApp value={ initialValue }/> )
        //fireEvent: es un método que nos permite disparar eventos en el DOM
        fireEvent.click( screen.getByText('+1') )

        expect( screen.getByText( '21' ) ).toBeTruthy()
    })

    test('Debe de disminuir -1', () => { 
        const { container } = render( <CounterApp value={ initialValue }/> )
        //fireEvent: es un método que nos permite disparar eventos en el DOM
        fireEvent.click( screen.getByText('-1') )

        expect( screen.getByText( '19' ) ).toBeTruthy()
    })

    test('Debe de resetear con el botón reset', () => { 
        const { container } = render( <CounterApp value={ 2312 }/> )
        //fireEvent: es un método que nos permite disparar eventos en el DOM
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('Reset') )
        // expect( screen.getByText(2312) ).toBeTruthy()

        //acceder al elemento con el aria-label
        screen.getByRole('button', {name: 'btn-reset'})
    })
})