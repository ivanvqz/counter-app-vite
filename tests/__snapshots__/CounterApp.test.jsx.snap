// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="center container"
  >
    <h1
      class="color"
    >
      CounterApp
    </h1>
    <h2>
       
      20
       
    </h2>
    <div>
      <button>
        +1
      </button>
      <button>
        -1
      </button>
      <button
        aria-label="btn-reset"
      >
        Reset
      </button>
    </div>
  </div>
</div>
`;
